zebrad/src/components/inbound.rs

```diff
@@ -171,9 +197,25 @@ impl Service<zn::Request> for Inbound {
                 debug!("ignoring unimplemented request");
                 async { Ok(zn::Response::Nil) }.boxed()
             }
-            zn::Request::AdvertiseBlock(_block) => {
-                debug!("ignoring unimplemented request");
-                async { Ok(zn::Response::Nil) }.boxed()
+            zn::Request::AdvertiseBlock(hash) => {
+                // this sucks
+                let mut downloads = self.downloads.take().unwrap();
+                self.downloads = Some(Downloads::new(
+                    self.outbound.as_ref().unwrap().clone(),
+                    self.verifier.clone(),
+                    self.state.clone(),
+                ));
```

Comment:
The lifetime of `self.downloads` is giving problems here so i made that dirty stuff above. It needs to be addressed properly.